python/test/unit/language/test_core.py

```diff
@@ -2460,6 +2460,26 @@ def histogram_kernel(x_ptr, z_ptr, M: tl.constexpr, N: tl.constexpr):
     histogram_kernel[(1, )](x, z, M=M, N=N)
     assert (z_torch == z).all()
 
+@pytest.mark.interpreter
+@pytest.mark.parametrize("M, N", [[2048, 2], [1024, 8], [1024, 128], [256, 512], [32, 512], [8, 512], [8, 2]])
+def test_histogram_broadcast(M, N, device):
+    
+    @triton.jit
+    def histogram_kernel_broadcast(x_ptr, z_ptr, M: tl.constexpr, N: tl.constexpr):
+        offset1 = tl.arange(0, M)
+        offset2 = tl.arange(0, N)
+        x = tl.load(x_ptr + offset1)
+        z = tl.histogram(x, N)
+        bias = tl.full([M, N], 1, dtype=tl.int32)
+        biased = z + bias
+        tl.store(z_ptr + offset1[:, None] + offset2[None, :], biased)
+
+    torch.manual_seed(17)
+    x = torch.randint(0, N, (M, ), device=device, dtype=torch.int32)
+    z = torch.empty(N, dtype=torch.int32, device=device)
+    z_torch = torch.histc(x.float(), bins=N, min=0, max=N - 1)
+    histogram_kernel_broadcast[(1, )](x, z, M=M, N=N)
+    assert (z_torch == z).all()
```

Comment:
what is this testing that wasn't tested before?

Reply:
The new test tests the situation when broadcast is done after histogram